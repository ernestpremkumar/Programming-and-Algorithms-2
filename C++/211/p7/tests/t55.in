balanced
print
breadth
echo 1
print
breadth
size
find 3
breadth
breadth
echo 2
balanced
print
insert 5
balanced
echo 0
distance
breadth
echo 6
find 3
echo 4
print
find 3
distance
balanced
find 4
balanced
echo 2
balanced
balanced
insert 6
size
breadth
breadth
size
size
insert 3
distance
breadth
breadth
echo 4
print
size
size
breadth
echo 2
insert 6
breadth
insert 6
print
find 4
balanced
print
echo 5
insert 1
distance
distance
insert 2
distance
find 4
distance
insert 3
distance
distance
echo 7
find 2
echo 3
balanced
balanced
breadth
distance
find 3
breadth
breadth
echo 3
distance
print
distance
breadth
insert 4
distance
insert 5
balanced
size
insert 7
echo 7
find 7
find 3
balanced
print
insert 0
distance
size
balanced
echo 7
breadth
distance
print
echo 5
print
distance
echo 4
balanced
insert 3
print
balanced
echo 6
find 5
size
insert 3
print
insert 7
distance
balanced
print
balanced
echo 2
find 7
size
balanced
size
breadth
find 2
print
find 5
echo 5
print
insert 1
distance
echo 1
breadth
print
distance
breadth
print
echo 4
size
echo 5
breadth
find 4
insert 1
echo 3
print
echo 2
find 3
breadth
echo 0
breadth
insert 7
echo 3
balanced
find 0
breadth
echo 3
balanced
find 3
balanced
balanced
distance
distance
echo 1
insert 4
distance
echo 6
insert 7
breadth
echo 1
find 0
print
balanced
distance
print
balanced
insert 3
find 5
distance
find 3
print
breadth
insert 4
print
echo 6
distance
insert 3
distance
distance
print
size
distance
echo 3
size
print
breadth
balanced
breadth
insert 0
distance
distance
print
balanced
insert 2
balanced
echo 5
echo 1
print
balanced
distance
print
print
breadth
print
print
echo 3
echo 2
size
balanced
breadth
echo 6
balanced
find 0
echo 3
size
breadth
insert 6
size
print
distance
size
size
find 2
insert 5
echo 3
insert 6
find 1
insert 1
print
echo 2
echo 6
breadth
echo 2
insert 7
distance
print
breadth
size
find 1
print
size
breadth
distance
print
echo 3
insert 5
find 1
find 6
distance
balanced
find 4
breadth
insert 0
insert 7
find 5
insert 5
find 6
print
print
print
breadth
distance
print
find 1
breadth
balanced
breadth
echo 4
find 7
print
balanced
breadth
find 2
print
insert 7
print
distance
balanced
find 3
breadth
breadth
echo 4
echo 3
size
find 4
find 6
find 2
breadth
insert 6
breadth
insert 2
insert 2
print
find 1
print
echo 0
breadth
print
size
insert 6
print
balanced
print
breadth
breadth
echo 3
breadth
balanced
size
insert 2
size
echo 2
find 7
breadth
find 5
size
find 7
balanced
distance
breadth
balanced
size
distance
distance
find 1
balanced
insert 6
breadth
print
insert 2
insert 7
distance
breadth
find 5
size
distance
breadth
distance
size
breadth
find 6
print
print
insert 1
print
balanced
distance
insert 5
find 3
distance
breadth
distance
distance
distance
echo 0
distance
echo 4
insert 2
print
print
breadth
breadth
echo 4
balanced
size
echo 6
find 7
size
find 1
insert 0
print
echo 0
size
echo 1
find 6
distance
balanced
breadth
insert 7
find 4
size
find 3
size
distance
distance
size
echo 2
distance
find 2
insert 6
size
balanced
find 1
echo 4
insert 4
echo 3
balanced
insert 4
insert 1
balanced
balanced
insert 1
find 7
size
breadth
breadth
size
echo 4
insert 5
size
print
breadth
distance
breadth
distance
balanced
distance
breadth
echo 1
echo 7
size
breadth
insert 4
size
balanced
print
distance
balanced
echo 5
breadth
echo 6
print
breadth
echo 1
echo 6
print
breadth
distance
balanced
insert 0
print
size
balanced
echo 1
breadth
distance
insert 5
print
find 7
breadth
print
breadth
size
balanced
print
echo 0
distance
print
find 5
breadth
insert 0
distance
distance
insert 2
distance
breadth
balanced
find 3
balanced
print
find 3
print
balanced
distance
print
find 4
echo 5
echo 1
find 6
echo 1
echo 1
find 6
insert 7
insert 6
echo 0
find 5
size
print
echo 1
find 3
balanced
size
balanced
breadth
print
find 6
print
find 6
print
breadth
size
echo 0
echo 4
breadth
echo 1
distance
balanced
distance
find 7
insert 3
insert 7
find 5
find 5
size
find 0
breadth
echo 6
size
size
insert 0
insert 2
insert 7
echo 0
balanced
find 3
size
distance
size
find 1
insert 4
print
breadth
print
print
balanced
insert 6
breadth
insert 2
size
echo 0
insert 5
insert 2
balanced
insert 3
echo 3
size
insert 4
echo 7
distance
find 4
size
distance
balanced
breadth
echo 1
distance
echo 1
insert 7
find 0
find 2
print
balanced
print
size
find 2
echo 6
size
distance
distance
size
insert 6
echo 0
echo 2
size
find 3
distance
size
breadth
print
echo 7
breadth
breadth
echo 0
insert 3
balanced
breadth
print
size
distance
balanced
insert 1
echo 5
size
insert 5
breadth
distance
find 2
size
balanced
insert 4
echo 6
distance
find 4
insert 1
breadth
size
distance
size
echo 0
balanced
distance
echo 5
echo 1
print
distance
breadth
print
find 4
breadth
insert 4
print
insert 3
distance
find 4
echo 7
print
balanced
balanced
balanced
balanced
echo 5
print
echo 4
print
find 5
echo 5
echo 7
size
breadth
distance
balanced
find 2
balanced
breadth
breadth
find 4
insert 6
size
distance
balanced
find 1
insert 7
echo 4
find 3
echo 6
find 1
find 7
breadth
insert 2
echo 2
balanced
breadth
find 0
insert 7
distance
find 0
balanced
breadth
balanced
distance
insert 7
distance
echo 5
balanced
distance
insert 0
print
breadth
breadth
insert 6
balanced
breadth
insert 6
distance
find 1
echo 4
distance
insert 7
size
echo 1
insert 0
breadth
balanced
print
insert 5
echo 6
find 0
distance
echo 4
breadth
insert 3
balanced
breadth
distance
size
echo 6
echo 3
echo 5
echo 7
print
size
insert 5
size
size
print
breadth
breadth
breadth
distance
size
size
find 2
balanced
insert 3
balanced
echo 6
find 2
balanced
insert 5
echo 1
insert 5
find 5
size
insert 1
print
echo 3
breadth
echo 3
distance
print
echo 6
echo 4
distance
echo 7
insert 7
print
distance
echo 0
insert 0
insert 1
print
print
breadth
distance
distance
insert 2
distance
insert 1
insert 4
find 2
size
print
echo 0
insert 4
distance
distance
find 2
print
balanced
print
print
print
balanced
balanced
balanced
size
size
size
size
find 3
breadth
find 2
print
insert 7
insert 3
insert 5
find 5
balanced
size
size
insert 2
print
size
distance
size
breadth
find 7
echo 0
balanced
find 2
balanced
breadth
insert 2
print
find 4
distance
size
balanced
echo 4
insert 5
balanced
size
distance
size
print
breadth
size
distance
distance
echo 3
insert 1
breadth
echo 4
echo 5
size
find 1
balanced
print
balanced
size
print
echo 7
insert 7
find 6
insert 0
insert 3
insert 3
balanced
print
insert 7
insert 6
echo 4
insert 0
distance
size
breadth
echo 7
distance
find 1
insert 1
print
echo 3
size
distance
distance
print
print
echo 4
echo 5
distance
distance
size
echo 6
echo 0
insert 4
insert 1
print
size
echo 6
print
balanced
distance
breadth
insert 6
print
insert 7
find 2
insert 3
find 3
balanced
size
insert 0
insert 7
breadth
find 1
echo 2
find 2
distance
print
distance
distance
size
size
breadth
print
insert 4
breadth
distance
breadth
echo 3
breadth
find 0
distance
insert 1
echo 2
size
balanced
echo 4
echo 1
balanced
insert 4
find 0
find 6
breadth
breadth
insert 5
breadth
find 4
breadth
breadth
find 5
find 7
find 2
print
echo 3
echo 3
find 2
echo 3
breadth
balanced
size
print
print
size
find 1
distance
distance
insert 1
breadth
breadth
echo 5
balanced
insert 4
size
breadth
echo 1
insert 0
balanced
print
distance
insert 7
distance
distance
size
breadth
echo 7
breadth
size
echo 6
print
distance
breadth
echo 7
breadth
balanced
print
echo 0
echo 2
size
insert 2
size
distance
insert 7
breadth
print
balanced
breadth
size
insert 5
size
size
insert 6
echo 2
echo 1
print
find 1
find 6
breadth
insert 4
distance
find 5
echo 5
