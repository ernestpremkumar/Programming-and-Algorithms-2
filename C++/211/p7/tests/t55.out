Tree is balanced.
{}
{}
1
{}
{}
0
<3> is not in tree.
{}
{}
2
Tree is balanced.
{}
Tree is balanced.
0
Average distance of nodes to root = 0
{5}
6
<3> is not in tree.
4
{5}
<3> is not in tree.
Average distance of nodes to root = 0
Tree is balanced.
<4> is not in tree.
Tree is balanced.
2
Tree is balanced.
Tree is balanced.
2
{5, 6}
{5, 6}
2
2
Average distance of nodes to root = 0.666667
{5, 3, 6}
{5, 3, 6}
4
{3, 5, 6}
3
3
{5, 3, 6}
2
{5, 3, 6}
{3, 5, 6}
<4> is not in tree.
Tree is balanced.
{3, 5, 6}
5
Average distance of nodes to root = 1
Average distance of nodes to root = 1
Average distance of nodes to root = 1.4
<4> is not in tree.
Average distance of nodes to root = 1.4
Average distance of nodes to root = 1.4
Average distance of nodes to root = 1.4
7
<2> is in tree.
3
Tree is not balanced.
Tree is not balanced.
{5, 3, 6, 1, 2}
Average distance of nodes to root = 1.4
<3> is in tree.
{5, 3, 6, 1, 2}
{5, 3, 6, 1, 2}
3
Average distance of nodes to root = 1.4
{1, 2, 3, 5, 6}
Average distance of nodes to root = 1.4
{5, 3, 6, 1, 2}
Average distance of nodes to root = 1.5
Tree is not balanced.
6
7
<7> is in tree.
<3> is in tree.
Tree is balanced.
{1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
8
Tree is balanced.
7
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
5
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
4
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
6
<5> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
2
<7> is in tree.
8
Tree is balanced.
8
{5, 3, 6, 1, 4, 7, 0, 2}
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
5
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
1
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
4
8
5
{5, 3, 6, 1, 4, 7, 0, 2}
<4> is in tree.
3
{0, 1, 2, 3, 4, 5, 6, 7}
2
<3> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
0
{5, 3, 6, 1, 4, 7, 0, 2}
3
Tree is balanced.
<0> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
3
Tree is balanced.
<3> is in tree.
Tree is balanced.
Tree is balanced.
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
1
Average distance of nodes to root = 1.75
6
{5, 3, 6, 1, 4, 7, 0, 2}
1
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
<5> is in tree.
Average distance of nodes to root = 1.75
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
6
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.75
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Tree is balanced.
5
1
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
3
2
8
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
6
Tree is balanced.
<0> is in tree.
3
8
{5, 3, 6, 1, 4, 7, 0, 2}
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
8
8
<2> is in tree.
3
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
2
6
{5, 3, 6, 1, 4, 7, 0, 2}
2
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
8
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
3
<1> is in tree.
<6> is in tree.
Average distance of nodes to root = 1.75
Tree is balanced.
<4> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
<5> is in tree.
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
4
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
Tree is balanced.
<3> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
4
3
8
<4> is in tree.
<6> is in tree.
<2> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
0
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
3
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
8
8
2
<7> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
<5> is in tree.
8
<7> is in tree.
Tree is balanced.
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
8
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
<1> is in tree.
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
<5> is in tree.
8
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
8
{5, 3, 6, 1, 4, 7, 0, 2}
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.75
<3> is in tree.
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
0
Average distance of nodes to root = 1.75
4
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
4
Tree is balanced.
8
6
<7> is in tree.
8
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
0
8
1
<6> is in tree.
Average distance of nodes to root = 1.75
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
<4> is in tree.
8
<3> is in tree.
8
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
8
2
Average distance of nodes to root = 1.75
<2> is in tree.
8
Tree is balanced.
<1> is in tree.
4
3
Tree is balanced.
Tree is balanced.
Tree is balanced.
<7> is in tree.
8
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
8
4
8
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Tree is balanced.
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
1
7
8
{5, 3, 6, 1, 4, 7, 0, 2}
8
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
Tree is balanced.
5
{5, 3, 6, 1, 4, 7, 0, 2}
6
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
1
6
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is balanced.
1
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
8
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
0
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
<3> is in tree.
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
5
1
<6> is in tree.
1
1
<6> is in tree.
0
<5> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
1
<3> is in tree.
Tree is balanced.
8
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
8
0
4
{5, 3, 6, 1, 4, 7, 0, 2}
1
Average distance of nodes to root = 1.75
Tree is balanced.
Average distance of nodes to root = 1.75
<7> is in tree.
<5> is in tree.
<5> is in tree.
8
<0> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
6
8
8
0
Tree is balanced.
<3> is in tree.
8
Average distance of nodes to root = 1.75
8
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
8
0
Tree is balanced.
3
8
7
Average distance of nodes to root = 1.75
<4> is in tree.
8
Average distance of nodes to root = 1.75
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
1
Average distance of nodes to root = 1.75
1
<0> is in tree.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
<2> is in tree.
6
8
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
8
0
2
8
<3> is in tree.
Average distance of nodes to root = 1.75
8
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
7
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
0
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.75
Tree is balanced.
5
8
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
<2> is in tree.
8
Tree is balanced.
6
Average distance of nodes to root = 1.75
<4> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
8
Average distance of nodes to root = 1.75
8
0
Tree is balanced.
Average distance of nodes to root = 1.75
5
1
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
<4> is in tree.
7
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Tree is balanced.
Tree is balanced.
Tree is balanced.
5
{0, 1, 2, 3, 4, 5, 6, 7}
4
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
5
7
8
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Tree is balanced.
<2> is in tree.
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
<4> is in tree.
8
Average distance of nodes to root = 1.75
Tree is balanced.
<1> is in tree.
4
<3> is in tree.
6
<1> is in tree.
<7> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
2
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
<0> is in tree.
Average distance of nodes to root = 1.75
<0> is in tree.
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
5
Tree is balanced.
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
<1> is in tree.
4
Average distance of nodes to root = 1.75
8
1
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
6
<0> is in tree.
Average distance of nodes to root = 1.75
4
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
8
6
3
5
7
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
8
8
<2> is in tree.
Tree is balanced.
Tree is balanced.
6
<2> is in tree.
Tree is balanced.
1
<5> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
3
{5, 3, 6, 1, 4, 7, 0, 2}
3
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
6
4
Average distance of nodes to root = 1.75
7
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
0
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
<2> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
0
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Tree is balanced.
Tree is balanced.
8
8
8
8
<3> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
Tree is balanced.
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.75
8
{5, 3, 6, 1, 4, 7, 0, 2}
<7> is in tree.
0
Tree is balanced.
<2> is in tree.
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
Average distance of nodes to root = 1.75
8
Tree is balanced.
4
Tree is balanced.
8
Average distance of nodes to root = 1.75
8
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
8
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
3
{5, 3, 6, 1, 4, 7, 0, 2}
4
5
8
<1> is in tree.
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
7
<6> is in tree.
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
4
Average distance of nodes to root = 1.75
8
{5, 3, 6, 1, 4, 7, 0, 2}
7
Average distance of nodes to root = 1.75
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
3
8
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
4
5
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
8
6
0
{0, 1, 2, 3, 4, 5, 6, 7}
8
6
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
<3> is in tree.
Tree is balanced.
8
{5, 3, 6, 1, 4, 7, 0, 2}
<1> is in tree.
2
<2> is in tree.
Average distance of nodes to root = 1.75
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
8
8
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
3
{5, 3, 6, 1, 4, 7, 0, 2}
<0> is in tree.
Average distance of nodes to root = 1.75
2
8
Tree is balanced.
4
1
Tree is balanced.
<0> is in tree.
<6> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
<4> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
<5> is in tree.
<7> is in tree.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
3
3
<2> is in tree.
3
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
<1> is in tree.
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
{5, 3, 6, 1, 4, 7, 0, 2}
5
Tree is balanced.
8
{5, 3, 6, 1, 4, 7, 0, 2}
1
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
Average distance of nodes to root = 1.75
8
{5, 3, 6, 1, 4, 7, 0, 2}
7
{5, 3, 6, 1, 4, 7, 0, 2}
8
6
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
7
{5, 3, 6, 1, 4, 7, 0, 2}
Tree is balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
0
2
8
8
Average distance of nodes to root = 1.75
{5, 3, 6, 1, 4, 7, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is balanced.
{5, 3, 6, 1, 4, 7, 0, 2}
8
8
8
2
1
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
<6> is in tree.
{5, 3, 6, 1, 4, 7, 0, 2}
Average distance of nodes to root = 1.75
<5> is in tree.
5
